external-crates/move/crates/move-vm-runtime/src/tracing2/tracer.rs

```diff
@@ -532,9 +531,8 @@ impl VMTracer<'_> {
                     layout,
                     ref_type: ref_type
                         .map(|r_type| {
-                            let possible_location = start_trace_index + i;
-                            if self.loaded_data.contains_key(&possible_location) {
```

Comment:
Out of curiosity, could this ever be true in the previous version?

Reply:
It definitely could! It would work as long as all the references were at the beginning of the function and sequential E.g, if you had a function signature like
```move
public fun f(x: &T, ctx: &mut TxContext) 
```
Then it would be fine.

The issue arises when you have something like

```move
public fun f(x: &T, u: u64, ctx: &mut TxContext)
```
then this index would be all off.